src/test/java/uk/gov/justice/probation/courtcaseservice/listener/ProbationOffenderEventsListenerIntTest.java

```diff
@@ -0,0 +1,110 @@
+package uk.gov.justice.probation.courtcaseservice.listener;
+
+import com.amazonaws.services.sns.model.MessageAttributeValue;
+import com.amazonaws.services.sns.model.PublishRequest;
+import com.amazonaws.services.sns.model.PublishResult;
+import com.amazonaws.services.sqs.model.PurgeQueueRequest;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.springframework.boot.test.mock.mockito.SpyBean;
+import org.testcontainers.shaded.com.fasterxml.jackson.core.JsonProcessingException;
+import org.testcontainers.shaded.com.fasterxml.jackson.databind.ObjectMapper;
+import uk.gov.justice.probation.courtcaseservice.BaseIntTest;
+import uk.gov.justice.probation.courtcaseservice.jpa.entity.OffenderEntity;
+import uk.gov.justice.probation.courtcaseservice.jpa.entity.OffenderProbationStatus;
+import uk.gov.justice.probation.courtcaseservice.jpa.repository.OffenderRepository;
+import uk.gov.justice.probation.courtcaseservice.service.UserAgnosticOffenderService;
+import uk.gov.justice.probation.courtcaseservice.testUtil.OffenderEvent;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.Collections;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class ProbationOffenderEventsListenerIntTest extends BaseIntTest {
+
+    @SpyBean
+    UserAgnosticOffenderService offenderService;
+
+    @SpyBean
+    OffenderRepository offenderRepository;
+
+    ObjectMapper objectMapper;
+    @Captor
+    ArgumentCaptor<String> crnArgumentCaptor;
+
+    @Captor
+    ArgumentCaptor<OffenderEntity> offenderEntityArgumentCaptor;
+
+
+    @BeforeEach
+    public void setUp() {
+        objectMapper = new ObjectMapper();
+        getOffenderEventReceiverQueueSqsClient().purgeQueue(new PurgeQueueRequest(getOffenderEventReceiverQueueUrl()));
+    }
+
+    @Test
+    public void shouldProcess_OffenderEventChangedMessage_AndUpdateOffenderProbationStatus() throws JsonProcessingException, InterruptedException {
+        var offenderEntity = OffenderEntity.builder()
+                .id(Long.valueOf(-1000001))
+                .cro("CROINT007")
+                .crn("X320741")
+                .pnc("PNCINT007")
+                .probationStatus(OffenderProbationStatus.CURRENT)
+                .previouslyKnownTerminationDate(LocalDate.of(2010, 1, 1))
+                .awaitingPsr(true)
+                .breach(true)
+                .preSentenceActivity(true)
+                .build();
+
+        when(offenderRepository.findByCrn("X320741")).thenReturn(Optional.ofNullable(offenderEntity));
```

Comment:
Mocked this call (with the original details form the Db) due to OffenderRepository always returns an empty record.